project-koku/koku
# 1920 Sources: azure patch

#1920 Allow a PATCH for a single field of Azure billing-source.

Testing:
1. Add azure source.
2. PATCH either `resource_group` _or_ `storage_account`.
3. See successful PATCH.

```
{
    "id": 2,
    "uuid": "be36083c-1360-4dfb-aa0f-358d8953af27",
    "name": "mskarbek-azure",
    "source_type": "Azure",
    "authentication": {
        "credentials": {
            "client_id": "f36c251e-ad4a-4447-bfbb-fc9bb10f259a",
            "tenant_id": "ae4f8f55-f1a8-4080-9aa8-10779e4113e7",
            "client_secret": "pfy?Jk65.rQ4v*H9/CB]E7jtg[NA0mdX"
        }
    },
    "billing_source": {
        "data_source": {
            "resource_group": "RG1",
            "storage_account": "mysa3"
        }
    }
}
```
PATCH:
```
{
    "billing_source": {
        "data_source": {
        	"resource_group": "new-group"
        }
    }
}
```

Result:
```
{
    "id": 2,
    "uuid": "be36083c-1360-4dfb-aa0f-358d8953af27",
    "name": "mskarbek-azure",
    "source_type": "Azure",
    "authentication": {
        "credentials": {
            "client_id": "f36c251e-ad4a-4447-bfbb-fc9bb10f259a",
            "tenant_id": "ae4f8f55-f1a8-4080-9aa8-10779e4113e7",
            "client_secret": "pfy?Jk65.rQ4v*H9/CB]E7jtg[NA0mdX"
        }
    },
    "billing_source": {
        "data_source": {
            "resource_group": "new-group", <----------------
            "storage_account": "mysa3"
        }
    }
}
```
